app/helpers/smart_proxies_helper.rb

```diff
@@ -65,4 +65,25 @@ def tabbed_features(proxy)
   def show_feature_version(feature)
     render :partial => 'smart_proxies/plugins/plugin_version', :locals => { :feature => feature }
   end
+
+  def subnet_label(subnet)
+    foreman_subnet = Subnet.subnet_for(subnet.network)
+    if foreman_subnet
+      foreman_subnet.to_label
+    else
+      subnet.network
+    end
+  end
+
+  def link_to_host(subnet, hostname, smart_proxy)
+    foreman_subnet = Subnet.where(:network => subnet.network, :netmask => subnet.netmask)
+    return hostname unless foreman_subnet
+    host = hosts_with_subnet(subnet, smart_proxy, :name => hostname).first
+    host.present? ? link_to(host.to_s, host_path(host)) : hostname
+  end
+
+  def hosts_with_subnet(subnet, smart_proxy, condition = {})
+    subnets =  { :subnets => { :dhcp_id => smart_proxy.id, :network => subnet.network } }
+    Host.joins(:interfaces => :subnet).where(subnets.merge! condition)
```

Comment:
It was [there](https://github.com/xprazak2/foreman/commit/ab8b4a70969cf460ffd57b50a6373c7cebbe9964#diff-5b769784fd39cddb57cf503cd6f2392eR88) but somehow the earlier version got merged, creating a [fix](https://github.com/ohadlevy/foreman/compare/sp-dhcp...xprazak2:host-query?expand=1).